QEfficient/cloud/infer.py

```diff
@@ -45,102 +44,166 @@ def main(
-    logger.info(f"Model after Optimized transformations {model_transformed}")
-
-    # Export to the Onnx
-    logger.info(f"Exporting to Pytorch {model_name} to ONNX...")
-    base_path, generated_onnx_path = qualcomm_efficient_converter(
-        model_kv=model_transformed,
-        onnx_dir_path=onnx_dir_path,
-        model_name=model_name,
-        kv=True,
-        form_factor="cloud",
-        return_path=True,
-        tokenizer=tokenizer,
+    onnx_model_path = export_to_onnx(model_transformed, model_name, tokenizer, onnx_dir_path, onnx_model_path)
+    compile_and_execute_onnx(
+        onnx_model_path,
+        tokenizer,
+        batch_size,
+        qpc_dir_path,
+        num_cores,
+        prompt_len,
+        ctx_len,
+        mxfp6,
+        mxint8,
+        aic_enable_depth_first,
+        mos,
+        device_group,
+        prompt,
     )
-    print(
-        f"Generated Onnx_path {generated_onnx_path} and Onnx_model_path {onnx_model_path} and Onnx_dir_path is {onnx_dir_path}"
+
+
+def create_qpc_base_dir_name(
+    num_cores: int,
+    batch_size: int,
+    prompt_len: int,
+    ctx_len: int,
+    mos: int,
+    device_count: int,
+    mxfp6: bool,
+    mxint8: bool,
+) -> str:
+    """
+    Create the base directory name for the QPC based on the parameters.
+
+    Returns:
+    - A string representing the base directory name.
+    """
+    return f"qpc_{num_cores}cores_{batch_size}BS_{prompt_len}PL_{ctx_len}CL_{mos}MOS_{device_count}devices{'_mxfp6_mxint8' if mxfp6 and mxint8 else '_mxfp6' if mxfp6 else '_fp16_mxint8' if mxint8 else '_fp16'}"
+
+
+def process_prompt(prompt: str, prompts_txt_file_path: str, batch_size: int) -> str:
+    """
+    Process the prompt for text generation.
+
+    Returns:
+    - The processed prompt.
+    """
+    return check_batch_size_and_num_prompts(prompt, prompts_txt_file_path, batch_size)
+
+
+def load_tokenizer(model_hf_path: str) -> AutoTokenizer:
+    """
+    Load the tokenizer from the downloaded model.
+
+    Returns:
+    - An instance of the tokenizer.
+    """
+    return AutoTokenizer.from_pretrained(model_hf_path, use_cache=True, padding_side="left", trust_remote_code=True)
+
+
+def execute_with_qpc(
+    tokenizer: AutoTokenizer, batch_size: int, qpc_dir_path: str, device_group: List[int], prompt: str
+) -> None:
+    """
+    Execute the model with the pre-compiled QPC.
+    """
+    cloud_ai_100_exec_kv(
+        batch_size=batch_size, tokenizer=tokenizer, qpc_path=qpc_dir_path, device_id=device_group, prompt=prompt
     )
-    assert (
-        generated_onnx_path == onnx_model_path
-    ), f"ONNX files were generated at an unusual location, expected {onnx_model_path}, got {generated_onnx_path}"
-    logger.info(f"Base Path is {base_path} and Onnx Model Path is : {generated_onnx_path}")
 
-    # Compile
-    # We need to pass parent directory of qpc_dir_path, as the compile function handles the qpcs directory creation
+
+def compile_and_execute_onnx(
```

Comment:
do you think this function will be used by other files in future?

Reply:
yes, if user wants to have just `onnx `or compilation from `infer`, he can do it. instead of rotating around compile/export/ apis()

Reply:
what do you mean by "from `infer`"? Can we move these and `export_to_onnx` functions to `utils` file? and not keep it here? if it is going to be used by other files?